logging/logmanager/plugin_impl_log_manager.go

```diff
@@ -91,22 +93,40 @@ func (lm *Plugin) Init() error {
 		}
 		lm.Log.Debugf("logs config: %+v", lm.Conf)
 
+		// Handle default log level
 		if lm.Conf.DefaultLevel != "" {
 			if err := lm.LogRegistry.SetLevel("default", lm.Conf.DefaultLevel); err != nil {
-				lm.Log.Warn("setting default log level failed:", err)
+				lm.Log.Warnf("setting default log level failed: %v", err)
 			} else {
-				lm.Log.Debugf("default log level to %q", lm.Conf.DefaultLevel)
+				// All loggers created up to this point were created with initial log level set (defined
+				// via INITIAL_LOGLVL env. variable with value 'info' by default), so at first, let's set default
+				// log level for all of them.
+				for loggerName := range lm.LogRegistry.ListLoggers() {
+					logger, exists := lm.LogRegistry.Lookup(loggerName)
+					if !exists {
+						continue
+					}
+					logger.SetLevel(stringToLogLevel(lm.Conf.DefaultLevel))
```

Comment:
The truth is that the order of precedence is currently not defined for the vpp-agent, so you can't say what is or is not a correct behavior. I guess you got the order from [namsral/flag](https://github.com/namsral/flag#usage), but that's just their opinion, there is no higher logic in that. 

I personally prefer config file over environmental variables. The reason is simple, let say you have etcd.conf with endpoint and several other fields set in the config file, and ETCD_ENDPOINTS env var with a different value.  You'll end up with one field set via env var and other via config file, so it's a kind of a mess. (By the way, ETCD plugin implementation also prefers config file over env var).

Reply:
So I want to say we should agree on some order, define it in the readme and then update all implementation using both, env vars and config files, to follow it.

Reply:
Viper, which is how a large number of Go projects handle configs, uses the following order:

Viper uses the following precedence order. Each item takes precedence over the item below it:

* explicit call to Set
* flag
* env
* config
* key/value store
* default

Reply:
I'm ok with it. Also, I'm not saying that the order where env. var is preferred is problematic, only that it may be a bit confusing in some cases. If we agree on this precedence, I'll change it here - but I also want to point out that we should keep the order consistent in every plugin which uses env-var/conf-file combo together.